src/containers/auth/EmailPasswordSignUp.jsx

```diff
@@ -0,0 +1,141 @@
+import React from "react";
+import { Form, Button, Col, Container } from "react-bootstrap";
+import { Redirect } from "react-router-dom";
+import { connect } from "react-redux";
+import { signUp } from "../../actions/authActions";
+
+const EmailPasswordSignUp = (props) => {
+  const [firstName, setFirstName] = React.useState("");
+  const [lastName, setLastName] = React.useState("");
+  const [email, setEmail] = React.useState("");
+  const [password, setPassword] = React.useState("");
+  const [passwordRepeat, setPasswordRepeat] = React.useState("");
+  const [passwordMatch, setPasswordMatch] = React.useState(true);
+  const { auth, authError, errMessage, errKey } = props;
+  // Redirect to home page after it's signed up and logged in
+  if (auth.uid) {
+    return <Redirect to="/" />;
+  }
+  // Create a new account with "Email & Password" method
+  const handleSubmit = (e) => {
+    if (password === passwordRepeat) {
+      setPasswordMatch(true);
+      props.signUp({
+        firstName,
```

Comment:
still has the firstname thing?

Reply:
never mind I realized you are trying to collect both fields still but letting the user choose them. I forgot I suggested that earlier. pls ignore